lib/dialects/abstract/connection-manager.js

```diff
@@ -312,6 +288,36 @@ class ConnectionManager {
     if (!this.dialect.connectionManager.validate) return true;
     return this.dialect.connectionManager.validate(connection);
   }
+
+  _acquire(options) {
+    const connectionPromise = this.pool.acquire(options.priority, options.type, options.useMaster);
+    return this._watchPoolError(connectionPromise).tap(() => { debug('connection acquired'); });
+  }
+
+  _watchPoolError(acquirePromise) {
```

Comment:
It should still be possible without. What do you mean with "watching" the promise? If the promise sometimes stays pending forevern and you just need a timeout, that's just
```ts
Promise.race([p, new Promise(resolve => setTimeout(resolve, 5000))])
```